include/gauge_field_order.h

```diff
@@ -162,59 +158,6 @@ namespace quda {
 
   namespace gauge {
 
-    template<typename ReduceType, typename Float> struct square_ {
-      square_(ReduceType scale) { }
-      __host__ __device__ inline ReduceType operator()(const quda::complex<Float> &x)
-      { return static_cast<ReduceType>(norm(x)); }
-    };
-
-    template <typename ReduceType> struct square_<ReduceType, int8_t> {
-      const ReduceType scale;
-      square_(const ReduceType scale) : scale(scale) { }
-      __host__ __device__ inline ReduceType operator()(const quda::complex<int8_t> &x)
-      { return norm(scale * complex<ReduceType>(x.real(), x.imag())); }
-    };
-
-    template<typename ReduceType> struct square_<ReduceType,short> {
-      const ReduceType scale;
-      square_(const ReduceType scale) : scale(scale) { }
-      __host__ __device__ inline ReduceType operator()(const quda::complex<short> &x)
-      { return norm(scale * complex<ReduceType>(x.real(), x.imag())); }
-    };
-
-    template<typename ReduceType> struct square_<ReduceType,int> {
-      const ReduceType scale;
-      square_(const ReduceType scale) : scale(scale) { }
-      __host__ __device__ inline ReduceType operator()(const quda::complex<int> &x)
-      { return norm(scale * complex<ReduceType>(x.real(), x.imag())); }
-    };
-
-    template<typename Float, typename storeFloat> struct abs_ {
-      abs_(const Float scale) { }
-      __host__ __device__ Float operator()(const quda::complex<storeFloat> &x) { return abs(x); }
-    };
-
-    template <typename Float> struct abs_<Float, int8_t> {
-      Float scale;
-      abs_(const Float scale) : scale(scale) { }
-      __host__ __device__ Float operator()(const quda::complex<int8_t> &x)
-      { return abs(scale * complex<Float>(x.real(), x.imag())); }
-    };
-
-    template<typename Float> struct abs_<Float,short> {
-      Float scale;
-      abs_(const Float scale) : scale(scale) { }
-      __host__ __device__ Float operator()(const quda::complex<short> &x)
-      { return abs(scale * complex<Float>(x.real(), x.imag())); }
-    };
-
-    template<typename Float> struct abs_<Float,int> {
-      Float scale;
-      abs_(const Float scale) : scale(scale) { }
-      __host__ __device__ Float operator()(const quda::complex<int> &x)
-      { return abs(scale * complex<Float>(x.real(), x.imag())); }
-    };
-
     template <typename Float, typename storeFloat> __host__ __device__ inline constexpr bool fixed_point() { return false; }
```

Comment:
These were moved to reducer.h, so they only have to be defined once instead of in each accessor file.  I have just now added documentation to this file (50cc55380).